src/http_module.cpp

```diff
@@ -702,6 +718,40 @@ char* addResourceAttr(ngx_conf_t* cf, ngx_command_t* cmd, void* conf)
     return NGX_CONF_OK;
 }
 
+char *pemRootCertsRead(ngx_conf_t *cf, void *post, void *data) {
+    ngx_str_t *path = (ngx_str_t *)data;
+    auto mcf = getMainConf(cf);
+
+    if (path->len > 0) {
+        ngx_str_t full_path;
+        if (ngx_get_full_name(cf->pool, &full_path, &mcf->pemRootCerts) != NGX_OK) {
+            ngx_conf_log_error(NGX_LOG_EMERG, cf, 0, "Failed to get full path for %V", mcf->pemRootCerts);
+            return (char*)NGX_CONF_ERROR;
+        }
+
+        try {
+            std::ifstream file;
```

Comment:
We should keep using `is_open()` to check for opening error, so that we can log `errno` with `ngx_conf_log_error` directly. Exceptions thrown by fstream might be unhelpful.

It's OK to enable exceptions for subsequent operations as those are far less likely to fail.